src/modules/components/proposal/proposalDataListItem/proposalDataListItemStructure/proposalDataListItemStructure.tsx

```diff
@@ -0,0 +1,111 @@
+import classNames from 'classnames';
+import { type KeyboardEvent } from 'react';
+import { useAccount } from 'wagmi';
+import { DataList, Tag } from '../../../../../core';
+import { addressUtils } from '../../../../utils/addressUtils';
+import { ProposalDataListItemResult } from '../proposalDataListItemResult';
+import { ProposalDataListItemStatus } from '../proposalDataListItemStatus';
+import {
+    type IProposalDataListItemStructureProps,
+    type IProposalListItemBaseProps,
+} from './proposalDataListItemStructureApi';
+
+/**
+ * `ProposalDataListItemStructure` module component
+ */
+export const ProposalDataListItemStructure: React.FC<IProposalDataListItemStructureProps> = (props) => {
+    const { wagmiConfig: config, chainId, ...nonWeb3Props } = props;
+    const { address: connectedAddress, isConnected } = useAccount({ config });
+
+    let otherComponentProps;
+    let resultComponent;
+
+    // pick the proper props upfront so that only the valid base props
+    // are applied to the underlying DataListItem component
+    if (nonWeb3Props.type === 'approvalThreshold') {
+        const { approvalAmount, approvalThreshold, type, ...otherProps } = nonWeb3Props;
+        otherComponentProps = otherProps;
+
+        resultComponent = (
+            <ProposalDataListItemResult
+                approvalAmount={approvalAmount}
+                approvalThreshold={approvalThreshold}
+                type={type}
+            />
+        );
+    } else if (nonWeb3Props.type === 'majorityVoting') {
+        const { option, voteAmount, votePercentage, type, ...otherProps } = nonWeb3Props;
+        otherComponentProps = otherProps;
+
+        resultComponent = (
+            <ProposalDataListItemResult
+                option={option}
+                voteAmount={voteAmount}
+                votePercentage={votePercentage}
+                type={type}
+            />
+        );
+    }
```

Comment:
I would change the component props interface a bit to avoid such logic in place, when I see `let` initialisations inside a component I usually think that the code can be improved. 
Given that we already have a type property which tells us what to expect on the properties, we could have something like:
```
export interface IProposalListItemBaseProps<TType extends ProposalType = 'majorityVoting'>
    extends IDataListItemProps,
        IWeb3ComponentProps {
    [...]

    type: TType;
    result: TType extends 'majorityVoting' ? IMajorityVotingResult : IApprovalThresholdResult;
}
```
With this we won't have to destructure the props twice (`nonWeb3Props` & `otherProps`) and we could just pass to the result component the `result` and `type` properties. What do you think?